tests: bsim: bluetooth: mesh: Remove TinyCrypt bsim tests

PavelVPV opened the pull request:
This PR removes TinyCrypt Bluetooth mesh bsim tests.

PavelVPV committed:
tests: bsim: bluetooth: mesh: Remove bsim tests with TinyCrypt

As TinyCrypt gets deprecated, we need to remove all bsim tests that use
it. This commit switches all Bluetooth mesh bsim tests to PSA as a
default crypto backend.

This includes:
- removing `CONFIG_BT_MESH_USES_TINYCRYPT` as PSA is default crypto
  backend for mesh. `CONFIG_BT_MESH_USES_MBEDTLS_PSA` is not required to
  be enabled;
- merging `overlay_psa.conf` into `prj.conf` as this configuration
  becomes default for all mesh bsim tests;
- merging `overlay_ss.conf` into `overlay_pst.conf` as secure storage is
  required for running persistent storage bsim tests with PSA;

Signed-off-by: Pavel Vasilyev <pavel.vasilyev@nordicsemi.no>

alxelax commented:
If you want to test the low latency feature of Controller, please, implement bsim tests for Controller. Mesh is not correct place for testing this feature.

PavelVPV commented:
> If you want to test the low latency feature of Controller, please, implement bsim tests for Controller. Mesh is not correct place for testing this feature.

Could you show me where I added a new test for low latency feature in this PR?

alxelax reviewed (changes requested):
Testing of the low latency feature should go to ZLL bsim tests.

alxelax commented:
> > Testing of the low latency feature should go to ZLL bsim tests.
> 
> I disagree with this. This should be discussed separately. This PR is not about low latency feature at all.

This PR increases test scope on unnecessary manner. It should be done correctly. I will not approve such approach.

PavelVPV commented:
> > > Testing of the low latency feature should go to ZLL bsim tests.
> > 
> > 
> > I disagree with this. This should be discussed separately. This PR is not about low latency feature at all.
> 
> This PR increases test scope on unnecessary manner. It should be done correctly. I will not approve such approach.

It doesn't if you see the final diff except legacyt advertiser tests, same as yours.

PavelVPV commented:
> > > If you want to test the low latency feature of Controller, please, implement bsim tests for Controller. Mesh is not correct place for testing this feature.
> > 
> > 
> > Could you show me where I added a new test for low latency feature in this PR?
> 
> here: [3d813ab](https://github.com/zephyrproject-rtos/zephyr/commit/3d813ab93b67a7a16eb2b3838b0a0e0b1c0bdcb9)
> 
> all tests those are run with the low latency feature except transport and friendship are unnecessary

This is a preparation for switching to PSA.

alxelax commented:
> > > > If you want to test the low latency feature of Controller, please, implement bsim tests for Controller. Mesh is not correct place for testing this feature.
> > > 
> > > 
> > > Could you show me where I added a new test for low latency feature in this PR?
> > 
> > 
> > here: [3d813ab](https://github.com/zephyrproject-rtos/zephyr/commit/3d813ab93b67a7a16eb2b3838b0a0e0b1c0bdcb9)
> > all tests those are run with the low latency feature except transport and friendship are unnecessary
> 
> This is a preparation for switching to PSA.

It is why it should be done properly.

PavelVPV committed:
tests: bsim: bluetooth: mesh: Run low latency adv tests with PSA

This is a preparation commit for switching all bsim tests to PSA as
TinyCrypt will be removed soon.

Signed-off-by: Pavel Vasilyev <pavel.vasilyev@nordicsemi.no>

PavelVPV commented:
> I guess it is worth to split removing Tinycrypt and refactoring the low latency feature usage in different PRs to not block Tinycrypt deprecation procedure by not related discussion.

Yes, therefore I swithed low latency tests to PSA as well, so the scope doesn't change (except legacy advertiser tests w/o low lat feature).

omkar3141 commented:
> > > I guess it is worth to split removing Tinycrypt and refactoring the low latency feature usage in different PRs to not block Tinycrypt deprecation procedure by not related discussion.
> > 
> > 
> > Yes, therefore I swithed low latency tests to PSA as well, so the scope doesn't change (except legacy advertiser tests w/o low lat feature).
> 
> After adding legacy advertiser tests w/o low latency feature, all these tests with the low latency became obsolete and should be removed. Mesh is not the right place to test difference in ZLL Controller behavior with and without its features.

It is possible to simply create some light-weight advertiser test by calling host's advertiser APIs and test low latency using that.

PavelVPV commented:
> > > I guess it is worth to split removing Tinycrypt and refactoring the low latency feature usage in different PRs to not block Tinycrypt deprecation procedure by not related discussion.
> > 
> > 
> > Yes, therefore I swithed low latency tests to PSA as well, so the scope doesn't change (except legacy advertiser tests w/o low lat feature).
> 
> After adding legacy advertiser tests w/o low latency feature, all these tests with the low latency became obsolete and should be removed. Mesh is not the right place to test difference in ZLL Controller behavior with and without its features.

Different topic.

omkar3141 commented on tests/bsim/bluetooth/mesh/prj.conf:
```diff
@@ -74,5 +74,5 @@ CONFIG_BT_TESTING=y
-# Enable tinycrypt as a crypto backend
-CONFIG_BT_MESH_USES_TINYCRYPT=y
+# Increase the number of key slots in PSA Crypto core
+CONFIG_MBEDTLS_PSA_KEY_SLOT_COUNT=64
```
Side note: We will have to write recommendation on how to compute this number. When people start changing Netkey/Appkey indices this number will change, and unfortunately Kconfig parser does not do mathematical operations.

alxelax replied:
Unfortunately, static key slot allocation is available only here. Our Zephyr fork doesn't support it at the moment since different mbedTLS backend. They are allocated from heap.

omkar3141 reviewed (commented):
This looks ok to me. It will nice if we can take the low latency tests as a seperate discussion or PR, and let developers who want these tests to update them or refactor them to be not dependent on mesh.

PavelVPV commented:
Removed tests for legacy advertiser. Kept low latency tests.

alxelax commented on tests/bsim/bluetooth/mesh/compile.sh:
```diff
@@ -15,24 +15,16 @@ app=tests/bsim/bluetooth/mesh conf_overlay=overlay_pst.conf compile
 app=tests/bsim/bluetooth/mesh conf_overlay=overlay_workq_sys.conf compile
 app=tests/bsim/bluetooth/mesh conf_overlay=overlay_multi_adv_sets.conf compile
 app=tests/bsim/bluetooth/mesh conf_overlay=overlay_lpn_scan_on.conf compile
 app=tests/bsim/bluetooth/mesh \
```
I guess this line should be removed too. It doesn't throw error only because line 22 is empty.

alxelax commented on tests/bsim/bluetooth/mesh/overlay_pst.conf:
```diff
@@ -15,3 +15,6 @@ CONFIG_BT_MESH_SEQ_STORE_RATE=1
 CONFIG_BT_MESH_COMP_PST_BUF_SIZE=600
+
+# Required for PSA
+CONFIG_SECURE_STORAGE=y
```
Please fix commit message:
```
- merging `overlay_ss.conf` into `overlay_pst.conf` as secure storage is
  required for running persistent storage bsim tests with PSA;
```

alxelax reviewed (commented):
It looks much better but still requires minor fixes. Please take a look at my comments.

aescolar reviewed (commented):
From my POV I don't see any issue, so this can be considered a +1.
I do not know if you guys want any more reviews from the mesh side though.

aescolar reviewed (approved):
+1
